### DISCUSSION TITLE
Multiple parents

### DISCUSSION THREAD

**Discussion by tackITEasy** [ArcadeData/arcadedb#400] [Q&A] [answered]
Hi guys,

I'm new here.
I'm looking for a graphdb with the possibility to have a primary parent and secondaries parents.
Is this possible with ArcadeDB?

Thanks
Have a nice day

**Comment by lvca** [comment, marked as answer]
Hi @tackITEasy, yes you can. ArcadeDB allows the creation of bidirectional property graphs, so you can have a tree or more complicated structures:

1. Create an edge type 'ParentOf'
2. Create vertices and connect them with 'ParentOf' relationship. You can have as many as you want.
3. You could also create a 'PrimaryParentOf' edge type if you want to distinguish the kind of parent relationship

**Reply by tackITEasy** [reply]
Hi @Ivca,

This is wonderful.

Thanks you very much for this information.

Have a nice day

**Comment by tackITEasy** [comment]
Is there a limit I should not exceed for performances reasons ?
What are the performance if I have 100 parents by example ?
And some of the parents have up to 10 millions (or more) children ?

Thanks

**Reply by lvca** [reply]
With native graph databases, like ArcadeDB, Neo4j and OrientDB, crossing edges takes a constant time O(1), but if you have to cross millions of edges, then this O(1) * 1M. Your domain must be designed for your use cases. How do you mostly traverse your graph?

**Reply by tackITEasy** [reply]
In fact I want to build a graph where every elements (including the edges) are based on a RBAC (Role Access Base Control).
Permissions can be inherited from the parent element in the graph.
I didn't figure yet how to do that.

**Reply by lvca** [reply]
@tackITEasy if you model an edge type "Parent" and "Role", you can connect roles between resources. At runtime you can check if a resource is connected by filtering the role edges you're interested in. if not found, you can recursively do the same on the parent (if any).
